include/oneapi/dpl/internal/binary_search_extension_defs.h

```diff
@@ -25,6 +25,17 @@ namespace oneapi
 {
 namespace dpl
 {
+namespace internal
+{
+
+enum class search_algorithm
```

Comment:
Why only a forward declaration is not enough?
Like this?
`enum class search_algorithm;`

Reply:
I am not sure that this `enum` required at all: see https://github.com/oneapi-src/oneDPL/pull/1496#discussion_r1562193572

Reply:
I did not realize forward declaring enum classes were possible. Thanks for pointing that out. I have removed the changes to the binary search files and just forward declared the enum class.

Reply:
OK: forward declaration is enough too.
My idea was to split this brick into three separate bricks and remove this `enum`.
But I am fine with forward declaration too.